src/storage/compression.rs

```diff
@@ -0,0 +1,136 @@
+use failure::Error;
+use std::{collections::HashSet, fmt, io::Read};
+
+pub type CompressionAlgorithms = HashSet<CompressionAlgorithm>;
+
+macro_rules! enum_id {
+    ($vis:vis enum $name:ident { $($variant:ident = $discriminant:expr,)* }) => {
+        #[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+        $vis enum $name {
+            $($variant = $discriminant,)*
+        }
+
+        impl $name {
+            #[cfg(test)]
+            const AVAILABLE: &'static [Self] = &[$(Self::$variant,)*];
+        }
+
+        impl fmt::Display for CompressionAlgorithm {
```

Comment:
Bit of a nit, might as well use the fully qualified paths here and remove the external import